noether_filtering/src/submesh_extraction/extruded_polygon_submesh_extractor.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2018 Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <noether_filtering/submesh_extraction/extruded_polygon_mesh_extractor.h>
+
+#include <pcl/common/common.h>
+#include <pcl/segmentation/extract_polygonal_prism_data.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/ModelCoefficients.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/search/kdtree.h>
+#include <pcl/features/normal_3d.h>
+#include <pcl/pcl_config.h>
+
+namespace
+{
+#if PCL_VERSION_COMPARE(<, 1, 10, 0)
+template <typename T>
+using shared_ptr = boost::shared_ptr<T>;
+
+template <typename T>
+auto make_shared = boost::make_shared<T>;
+#else
+template <typename T>
+using shared_ptr = pcl::shared_ptr<T>;
+
+template <typename T>
+auto make_shared = pcl::make_shared<T>;
+#endif
+
+/** @brief Creates a shared pointer with no-op destructor for various PCL methods that require a pointer */
+template <typename T>
+shared_ptr<const T> createLocalPointer(const T& obj)
+{
+  return shared_ptr<const T>(&obj, [](const T*) {});
+}
```

Comment:
I think I misunderstood this at first glance.  Is this a tricky way of temporarily constructing a shared pointer 'wrapper' around an existing object?  I love it.  May be worthy of including in a separate header that can be used more broadly in Noether.

Unless you actually think this is hacky and hate it.  But I figure either make it broadly useful or shouldn't be used at all.

Reply:
I think it's okay. I'm pretty sure this is more or less what `pcl::shared_ptr` and `pcl::make_shared` are doing anyway. If we find other areas this is useful, I would be okay to move it into a common place so it can be used elsewhere. Let's do that in a separate PR where we want to re-use it somewhere else